lib/record_store/provider.rb

```diff
@@ -158,8 +158,7 @@ def retry_on_connection_errors(
             raise if max_retries <= 0
             max_retries -= 1
 
-            waiter.message = "Waiting to retry after receiving an unparseable response"
-            waiter.wait
+            waiter.wait("Waiting to retry after receiving an unparseable response")
```

Comment:
Recommending to use single quotes unless needing interpolations:

```suggestion
            waiter.wait('Waiting to retry after receiving an unparseable response')
```